### ISSUE TITLE
token: methods to decode account layouts for mint and token are not publicly accessible

### ISSUE THREAD

**Issue by mschneider** [solana-labs/solana-program-library#3217]
Currently all methods to access a token account or mint are coupled to a connection object. In order to work with data from other transport streams, e.g. when fetching accounts via geyser plugins or other indexers i can't use them and so just copied all the layout decoding logic. A simple fix like exporting unpack* wouldn't help either as it expects AccountInfo rather then a plain Buffer.

**Comment by joncinque** [comment]
Gotcha, can you provide the exact interface that you expect for this?  As you mention, in Rust we can't use the `AccountInfo` version since that's only on-chain, but maybe we just need to convert from `Account` https://github.com/solana-labs/solana/blob/3f47c83e5c646b1184a9937117d5399d935703c5/sdk/src/account.rs#L488 to `AccountInfo` and then unpack from that?

And in JS, true, we just need to have one more layer of indirection, ie there's `getMint` which only needs the connection for the first line, then operates on the `AccountInfo`.  So we can have something like:

```
export async function getMint(...) {
  const info = await connection.getAccountInfo(address, commitment);
  if (!info) throw new TokenAccountNotFoundError();
  return unpackMint(info);
}
export function unpackMint(...) {
  // all the other logic
}
```

How does that sound?

**Comment by mschneider** [comment]
no. account info is not necessarily available when the data has not been directly fetched from RPC. the unpack method right now has two concerns:
a) verify accountinfo
b) verify buffer contents

i'd like a method that only does b

**Comment by mschneider** [comment]
referring to this unpack method: https://github.com/solana-labs/solana-program-library/blob/3e9104736afc85465512eb11bb144920e31af418/token/js/src/state/account.ts#L160
but it should also apply to the proposed unpackMint

**Comment by joncinque** [comment]
At this point, it might be easiest for you to do a PR with the function signatures, or maybe even the whole thing :smile:

**Comment by joncinque** [comment]
I'm spending time in the area over the next couple of weeks, so now's the best time to get your requests in :wink:

**Comment by jordansexton** [comment]
This is published in `@solana/spl-token@0.3.1` now.
Please note that the arg order has changed to `unpackX(address, info, programId?)`
I also exposed an `unpackMultisig` function for parity.
